photon-ml/src/main/scala/com/linkedin/photon/ml/Params.scala

```diff
@@ -158,6 +160,9 @@ class Params {
    */
   var deleteOutputDirsIfExist: Boolean = false
 
+  var inputFormatType: InputFormatType = InputFormatType.AVRO
+  var featureDimension: Int = -1
```

Comment:
Okay, sounds reasonable. 

I was suggesting to count the number of unique features in ```IdentityIndexMapLoader``` when creating the map loader, which should be a computationally light job, but *not* to actually re-index the features as other map loaders do, because it's redundant for most LibSVM data set as you suggested. And I just want to be very cautious when introducing any user-facing API changes 

Please also note that ```featureDimension``` the user specified will also be used to decide the dimension of the dense vectors used in optimization (e.g., coefficients vector, gradient, hessian vector), so if the user specify an upper bound that is (much) larger than the actual number of unique features, Photon's performance may suffer.